presto-client/src/main/java/com/facebook/presto/client/QueryResults.java

```diff
@@ -339,4 +342,15 @@ private static Object fixValue(TypeSignature signature, Object value)
                 return value;
         }
     }
+
+    private static <T> T safeCast(Class<T> type, Object value)
+    {
+        if (!type.isInstance(value)) {
+            throw new ClientException(format(
+                    "Result data has wrong deserialized type (expected: %s, actual: %s)",
+                    type.getName(),
+                    value.getClass().getName()));
```

Comment:
Can `value == null` at this point?

Reply:
We never call this method with null. That's handled at the beginning of `fixValue()`